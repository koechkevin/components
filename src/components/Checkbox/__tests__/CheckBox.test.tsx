import { CheckboxProps } from 'antd/lib/checkbox';
import { mount, ReactWrapper } from 'enzyme';
import React from 'react';

import CheckBox from '../Checkbox';

describe('Tests for CheckBox', () => {
  const props: CheckboxProps = {
    onClick: jest.fn(),
  };

  const wrapper: ReactWrapper = mount(<CheckBox {...props} />);

  it('Tests rendering of CheckBox without crushing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should simulate checking of the checkBox', () => {
    const checkBoxInput: ReactWrapper = wrapper.find('.ant-checkbox-input').first();
    checkBoxInput.simulate('click');
    checkBoxInput.simulate('click');

    expect(props.onClick).toHaveBeenCalledTimes(2);
  });
});
