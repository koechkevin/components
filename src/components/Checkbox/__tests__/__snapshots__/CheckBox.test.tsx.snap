// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for CheckBox Tests rendering of CheckBox without crushing 1`] = `ReactWrapper {}`;
